src/settings.h

```diff
@@ -50,6 +51,24 @@ class CSettings : public QObject
         strFileName ( "" )
     {
         QObject::connect ( QCoreApplication::instance(), &QCoreApplication::aboutToQuit, this, &CSettings::OnAboutToQuit );
+
+#ifndef HEADLESS
+        // The Jamulus App will be created as either a QCoreApplication or QApplication (a subclass of QGuiApplication).
+        // State signals are only delivered to QGuiApplications, so we determine here whether we instantiated the GUI.
+        const QGuiApplication* pGApp = dynamic_cast<const QGuiApplication*> ( QCoreApplication::instance() );
+
+        if ( pGApp != nullptr )
+        {
+            QObject::connect ( pGApp, &QGuiApplication::saveStateRequest, this, [=] ( QSessionManager& ) { Save(); } );
```

Comment:
I looked up the signal `QGuiApplication::saveStateRequest`, and noticed it said

**Note: You should use [Qt::DirectConnection](https://doc.qt.io/qt-5/qt.html#ConnectionType-enum) when connecting to this signal.**

I'm not sure exactly what that means, but I wanted to check you'd considered it.

Reply:
OK, https://doc.qt.io/qt-5/qt.html#ConnectionType-enum shows what it means. I think since the receiver is a lambda and not a thread elsewhere, it will default to `Qt:DirectConnection` anyway. Would it be a good idea to specify it explicitly?

Reply:
No, I hadn't spotted that.  Also, I noticed the signal is defined inside a `#ifndef` in its header, so I wrapped the usage in the same.